packages/jest-fake-timers/src/jestFakeTimers.ts

```diff
@@ -432,19 +427,21 @@ export default class FakeTimers<TimerRef> {
       return null;
     }
 
-    const uuid = this._uuidCounter++;
+    const uuid = String(this._uuidCounter++);
 
     this._immediates.push({
       callback: () => callback.apply(null, args),
-      uuid: String(uuid),
+      uuid,
     });
 
-    const cancelledImmediates = this._cancelledImmediates;
+    const immediates = this._immediates;
     this._timerAPIs.setImmediate(() => {
-      if (!cancelledImmediates.hasOwnProperty(uuid)) {
-        // Callback may throw, so update the map prior calling.
-        cancelledImmediates[String(uuid)] = true;
-        callback.apply(null, args);
+      if (immediates.find(x => x.uuid === uuid)) {
```

Comment:
why do we need this `if`?

Reply:
please refer to this comment: https://github.com/facebook/jest/pull/8764#discussion_r312744317

Reply:
I mean, why do we have `find` at all?

Reply:
so we don't run immediates more than once.
If it was cleared (not in the `immediates`) then it shouldn't run even if `runAllImmediates` is called

Reply:
Right, makes sense